source/tutorials/module-system/module-system.md

```diff
@@ -0,0 +1,1289 @@
+# Deep dive demo: Wrapping the world in modules
+
+Much of the power in Nixpkgs and NixOS comes from the module system.
+It provides mechanisms for conveniently declaring and automatically merging interdependent attribute sets that follow dynamic type constraints, making it easy to express modular configurations.
+
+## Overview
+
+This tutorial follows [@infinisil](https://github.com/infinisil)'s [presentation on modules](https://infinisil.com/modules.mp4)  ([source](https://github.com/tweag/summer-of-nix-modules)) for  participants of [Summer of Nix](https://github.com/ngi-nix/summer-of-nix) 2021.
```

Comment:
Fyi I just updated https://infinisil.com/modules.mp4 to use H.264 instead, so this should now easily play in browsers :)